TYPO3.Flow/Classes/TYPO3/Flow/Property/TypeConverter/DateTimeConverter.php

```diff
@@ -138,13 +139,14 @@ public function convertFrom($source, $targetType, array $convertedChildPropertie
             }
             if (isset($source['dateFormat']) && strlen($source['dateFormat']) > 0) {
                 $dateFormat = $source['dateFormat'];
+                $isFormatSpecified = true;
             }
         }
         if ($dateAsString === '') {
             return null;
         }
-        $configurationForm = $configuration === null ? null : $configuration->getConfigurationValue('TYPO3\Flow\Property\TypeConverter\DateTimeConverter', self::CONFIGURATION_DATE_FORMAT);
-        if (ctype_digit($dateAsString) && $configurationForm === null && (!is_array($source) || !isset($source['dateFormat']))) {
+        $isFormatConfigured = ($configuration !== null && $configuration->getConfigurationValue(self::class, self::CONFIGURATION_DATE_FORMAT) !== null);
```

Comment:
Careful: `self:class` won't work if the class is proxied!. `DateTimeConverter::class` should do the trick